lib/mongo_compat_api.js

```diff
@@ -0,0 +1,280 @@
+var Promise = require("bluebird");
+var assert = require("assert");
+var fhditcher = require("./ditcher");
+var logger = require("./logger");
+var utils = require("./utils");
+var MongoClient = require("mongodb").MongoClient;
+
+/**
+ * Create a ditcher instance in case we are working with a shared
+ * database.
+ *
+ * @param mongoUrl Mongodb connection string to be used
+ * @param callback result callback
+ */
+function createDitcherInstance(mongoUrl, callback) {
+  // Default config used by ditcher if no connection string
+  // is provided
+  var config = {
+    database: {
+      host: '127.0.0.1',
+      port: process.env.FH_LOCAL_DB_PORT || 27017,
+      name: 'FH_LOCAL'
+    }
+  };
+
+  if (mongoUrl) {
+    try {
+      config = utils.parseMongoConnectionURL(mongoUrl);
+    } catch(e) {
+      return callback(e);
+    }
+  }
+
+  var versString = (mongoUrl) ? "db per app" : "shared db";
+  var ditcher = new fhditcher.Ditcher(config, logger, versString, function () {
+    return callback(null, ditcher);
+  });
+}
+
+/**
+ * Implements a MongoDB Node.js 2.0 driver compatible API that works for
+ * both, shared and dedicated database.
+ *
+ * @param ditcher ditcher instance
+ * @param params default params for the current app
+ * @constructor
+ */
+var MongoCompatApi = function (params) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This function has 194 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Function has a complexity of 43 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)